### ISSUE TITLE
add_vertex! API

### ISSUE THREAD

**Issue by filchristou** [JuliaGraphs/MetaGraphsNext.jl#41]
Some discussions (https://github.com/JuliaGraphs/Graphs.jl/issues/122, https://github.com/JuliaGraphs/Graphs.jl/issues/146, https://github.com/JuliaGraphs/Graphs.jl/issues/165) argue that `add_vertex!` should return the index of the newly added vertex. Currently, the compromise is that `add_vertex!` returns `Bool`. 

Would you consider entering a transient phase and support both ?

For example, this would mean appropriately modifying the following line:
https://github.com/JuliaGraphs/MetaGraphsNext.jl/blob/71cf31d8710fa419896af3ae03b9d4f943844402/src/graphs.jl#L117
A valid alternative would be to check `nv(meta_graph.graph)` before and after the `add_vertex!` operation.

### Motivation
As mentioned in https://github.com/JuliaGraphs/MetaGraphsNext.jl/pull/40 and https://github.com/QuantumBFS/Multigraphs.jl/pull/17 I would like to generalize the use cases for MetaGraphsNext and more specifically have it work with Multigraphs. 
`Multigraphs.jl`'s `add_vertex!` already return an index.

**Comment by bramtayl** [comment]
I'm indifferent here. If I had designed it, I'd have it return `nothing`, because users can also just use `nv` to find the new number of vertices, and error if the vertex wasn't added. But if others find it useful, sure, we can return the new index.

**Comment by filchristou** [comment]
MetaGraphsNext.jl is actually a wrapper, so I wouldn't try to impose a new style. `add_vertex!(mg::MetaGraphsNext.MetaGraph)` should return whatever `add_vertex!(mg.graph)` returns. The best we can do, is at least handle the API internally in a neutral fashion. Instead of doing the check `if add_vertex!(mg.graph) `do the check 
```julia
nvnum = nv(mg.graph)
add_vertex!(mg.graph)
if nv(mg.graph) != nvnum ...
```
If you agree with this changes, let me know and I can start working on a PR to substitute such checks.

**Comment by bramtayl** [comment]
Sure that sounds good thanks!

**Comment by gdalle** [comment]
> If you agree with this changes, let me know and I can start working on a PR to substitute such checks.

I also agree with this :)

**Comment by filchristou** [comment]
can you please assign this issue on me, so that I don't forget this ?

**Comment by gdalle** [comment]
Fixed by #47
